distributed/ddp-tutorial-series/multigpu_torchrun.py

```diff
@@ -11,8 +11,21 @@
 
 
 def ddp_setup():
-    torch.cuda.set_device(int(os.environ["LOCAL_RANK"]))
-    init_process_group(backend="nccl")
+    rank = int(os.environ["LOCAL_RANK"])
+    if torch.accelerator.is_available():
+        device_type = torch.accelerator.current_accelerator()
+        device: torch.device = torch.device(f"{device_type}:{rank}")
+        torch.accelerator.device_index(rank)
+        print(f"Running on rank {rank} on device {device}")
```

Comment:
I have hard time to understand this code block. It does not make sense to me in multiple places. Why you name what `current_accelerator()` as `device_type` if you return it from `ddp_setup()` in the same way as you return `device` for CPU path? Does `ddp_setup()` return different values? Next something is happening with the rank which is also not quite clear.

I think what you are trying to achieve is closer to this:
```suggestion
        torch.accelerator.set_device_index(rank)
        device = torch.accelerator.current_accelerator()
        print(f"Running on rank {rank} on device {device}")
```